pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -315,11 +358,17 @@ fi
 
 tmpDir=$(mktemp -t -d nixos-rebuild.XXXXXX)
 SSHOPTS="$NIX_SSHOPTS -o ControlMaster=auto -o ControlPath=$tmpDir/ssh-%n -o ControlPersist=60"
+if [ -n "$buildHost" -a -n "$flake" ]; then
+    buildHostCmd mkdir -p "$tmpDir"
+fi
 
 cleanup() {
     for ctrl in "$tmpDir"/ssh-*; do
         ssh -o ControlPath="$ctrl" -O exit dummyhost 2>/dev/null || true
     done
+    if [ -n "$buildHost" -a -n "$flake" ]; then
+        buildHostCmd rm -rf "$tmpDir"
```

Comment:
```
+ ssh -o ControlMaster=auto -o ControlPath=/tmp/nixos-rebuild.fEtksJ/ssh-%n -o ControlPersist=60 monitor sudo -- nix-env -p /nix/var/nix/profiles/system --set /tmp/nixos-rebuild.fEtksJ/result
error: getting status of '/tmp/nixos-rebuild.fEtksJ/result': No such file or directory
+ cleanup
+ for ctrl in "$tmpDir"/ssh-*
+ ssh -o ControlPath=/tmp/nixos-rebuild.fEtksJ/ssh-monitor -O exit dummyhost
+ '[' -n monitor -a -n git+file:///home/asymmetric/code/foo ']'
+ buildHostCmd rm -rf /tmp/nixos-rebuild.fEtksJ
+ '[' -z monitor ']'
+ '[' -n '' ']'
+ ssh -o ControlMaster=auto -o ControlPath=/tmp/nixos-rebuild.fEtksJ/ssh-%n -o ControlPersist=60 monitor sudo -- rm -rf /tmp/nixos-rebuild.fEtksJ
```

Note the `ControlPath=/tmp/nixos-rebuild.fEtksJ/ssh-%n`, rather than `/tmp/nixos-rebuild.fEtksJ/ssh-monitor` . It asked for the SSH key's password again, likely because of the call to `-O exit` above.

Reply:
Ah yes, the order should be switched here.

Reply:
Should be fixed.